### ISSUE TITLE
Duckhunter HID Letter"G" not working

### ISSUE THREAD

**Issue by j3223** [offensive-security/kali-nethunter#1052]
### Device:
Nexus 5x bullhead

### OS version (KitKat/Lollipop/Marshmallow/Nougat):
Nougat 7.1.2(latest Build)

### Built from repo (date and build command) or downloaded from website (links):
A user in a former issue of mine linked the newer custom built called nethunter-bullhead-nougat-kalifs-full-2.0.zip

### Output of `cat /proc/version` in adb shell or NetHunter Terminal:
Linux version (Linaro GCC 4.9-2017.01) ) #1 SMP PREEMPT Fri Nov 10 14:39:55 EST 2017

## Issue:
Hello. I began experimenting with the duckhunter HID attacks, but sadly found out, that the letter "G" or "g" wasnt outputted to the PC when I used the STRING command. All the other letters are working fine. The language is turned to german, cause my device is in german, but the issue remained when I turned the language to british english. Only in american english the letter was printed.
Any help or tips?
Thank you!

**Comment by bozwald** [comment]
So the letter d? Doesn't get outputted as text? When I first saw your other ticket I ran a hello world test on notepad and saw hello world but I'm nearly home so I'll try again and update

**Comment by j3223** [comment]
The hello world test also doesnt output a "d" or"D", but all other letters work. I really appreciate your help. Thx

**Comment by j3223** [comment]
And the problem is only with the german language selected, american english works...

**Comment by bozwald** [comment]
So the d works when you change your input to English with the zip I linked
you?

**Comment by j3223** [comment]
I did some more tests and apparently it is the letter "G" and "g", I dont know how I could have messed it up... I have installed the linked zip on a fresh install of the newest 7.1.2 built from google. If the language is german(my own language) or british english no "g" is sent to the pc. In american english all the letters work, but I cant use this language because all the symbols are different on an english keyboard. Ill change the title quick...

**Comment by bozwald** [comment]
REM This is a comment
STRING Hello world!
STRING Example of typing to computer. Nethunter is awesome!
REM To sleep for five seconds use miliseconds
SLEEP 5000
STRING I slept for 5 seconds, now I'm awake!
STRING abcdefghijklmnopqrstuvwxyz
STRING ABCDEFGHIJKLMNOPQRSTUVWXYZ
STRING 1234567890-=!@#$%^&*()_+
STRING []\;',./{}|:"<>?`~
MOUSE 300 300
ENTER
All of this script works for me on British English keyboard

**Comment by j3223** [comment]
![nethunterduckhuntergerror](https://user-images.githubusercontent.com/33799588/33088723-c0488372-ceee-11e7-914a-0e643f94d3da.PNG)

I ran the default hello world preset u linked...

**Comment by bozwald** [comment]
<img width="585" alt="screen shot 2017-11-21 at 18 27 41" src="https://user-images.githubusercontent.com/26123637/33089816-c20d4d46-cee9-11e7-990f-cad470749193.png">

![screenshot_20171121-183018](https://user-images.githubusercontent.com/26123637/33089978-4bc0f48e-ceea-11e7-9475-9752777536d2.png)

Its working for me bud. Are you using the Gboard as input method or Hacker's Keyboard? try Gboard to match me completely, I only use Hackers Keyboard in the terminal because I always end up hitting the full stop instead of a space bar and type.like.this :P

**Comment by bozwald** [comment]
Can you also try using the HID Keyboard, USB Keyboard app, and can you type the letters keyboard a-z and tell me if they're all working?

**Comment by j3223** [comment]
Im back and ill try both tipps...
BTW I used gboard

**Comment by j3223** [comment]
I tried changing the languages to the exact same as shown in the screenshot (british english as main), but it didnt help... The usb keyboard app works perfectly fine for all letters and numbers, but I think it has the american layout anyways...

**Comment by j3223** [comment]
What do i have to type into the terminal to do so?

**Comment by bozwald** [comment]
with debugging enabled in dev options on the phone

presuming you've adb and fastboot installed on your computer

adb logcat

heres a guide if u need help with that
https://forum.xda-developers.com/showthread.php?t=2303834

**Comment by j3223** [comment]
I did it, but the attack didnt work while usb debugging was enabled, so I had to restart the device for it to work again. Here is the logcat log, I have only copied the part when I pressed the button, because it loggs everything and it was becoming very much text. So here itis:
11-21 20:32:40.137  7799  7819 I PlayCommon: [30] com.google.android.play.a.h.e(248): Preparing logs for uploading
11-21 20:32:40.239  7799  7819 I PlayCommon: [30] com.google.android.play.a.h.a(470): Connecting to server: https://play.googleapis.com/play/log?format=raw&proto_v2=true
11-21 20:32:40.246  7799  7819 E PlayCommon: [30] com.google.android.play.a.h.a(500): Failed to connect to server: java.net.UnknownHostException: Unable to resolve host "play.googleapis.com": No address associated with hostname
11-21 20:32:45.388  9733 12782 D LANGGG  : de
11-21 20:32:45.441   370 12785 D Magisk  : su: request from client: 22
11-21 20:32:45.441   370 12785 D Magisk  : su: request from uid=[10094] (#1)
11-21 20:32:45.443   370 12785 D Magisk  : su_db: query policy=[2]
11-21 20:32:45.444   370 12785 D Magisk  : su: waiting child: [12786]
11-21 20:32:45.444 12786 12786 D Magisk  : su: child process started
11-21 20:32:45.444 12786 12786 D Magisk  : su: argc=[1]
11-21 20:32:45.444 12786 12786 D Magisk  : su: argv[0]=[su]
11-21 20:32:45.444 12786 12786 D Magisk  : su: cwd=[/]
11-21 20:32:45.444 12786 12786 D Magisk  : su: pts_slave=[]
11-21 20:32:45.445 12786 12786 D Magisk  : su: use namespace of pid=[12783]
11-21 20:32:45.485   370 12789 D Magisk  : su: request from client: 13
11-21 20:32:45.485   370 12789 D Magisk  : su: request from uid=[0] (#1)
11-21 20:32:45.487   370 12789 D Magisk  : su: waiting child: [12790]
11-21 20:32:45.487 12790 12790 D Magisk  : su: child process started
11-21 20:32:45.487 12790 12790 D Magisk  : su: argc=[3]
11-21 20:32:45.488 12790 12790 D Magisk  : su: argv[0]=[su]
11-21 20:32:45.488 12790 12790 D Magisk  : su: argv[1]=[-c]
11-21 20:32:45.488 12790 12790 D Magisk  : su: argv[2]=[/data/data/com.offsec.nethunter/files/scripts/bootkali duck-hunt-convert de /sdcard/nh_files/modules/duckconvert.txt /opt/duckout.sh]
11-21 20:32:45.488 12790 12790 D Magisk  : su: cwd=[/]
11-21 20:32:45.488 12790 12790 D Magisk  : su: pts_slave=[]
11-21 20:32:45.488 12790 12790 D Magisk  : su: use namespace of pid=[12788]
11-21 20:32:45.594   370 12795 D Magisk  : su: request from client: 21
11-21 20:32:45.594   370 12795 D Magisk  : su: request from uid=[0] (#2)
11-21 20:32:45.595   370 12795 D Magisk  : su: waiting child: [12796]
11-21 20:32:45.595 12796 12796 D Magisk  : su: child process started
11-21 20:32:45.595 12796 12796 D Magisk  : su: argc=[4]
11-21 20:32:45.595 12796 12796 D Magisk  : su: argv[0]=[su]
11-21 20:32:45.595 12796 12796 D Magisk  : su: argv[1]=[-c]
11-21 20:32:45.595 12796 12796 D Magisk  : su: argv[2]=[setenforce]
11-21 20:32:45.595 12796 12796 D Magisk  : su: argv[3]=[0]
11-21 20:32:45.595 12796 12796 D Magisk  : su: cwd=[/]
11-21 20:32:45.596 12796 12796 D Magisk  : su: pts_slave=[]
11-21 20:32:45.596 12796 12796 D Magisk  : su: use namespace of pid=[12794]
11-21 20:32:45.621   370 12795 D Magisk  : su: return code: [0]
11-21 20:32:45.780 12787 12787 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 0 <<<<<<
11-21 20:32:45.783 12787 12787 D AndroidRuntime: CheckJNI is OFF
11-21 20:32:45.821 12791 12791 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 0 <<<<<<
11-21 20:32:45.824 12791 12791 D AndroidRuntime: CheckJNI is OFF
11-21 20:32:45.864 12787 12787 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
11-21 20:32:45.900 12787 12787 I Radio-JNI: register_android_hardware_Radio DONE
11-21 20:32:45.911 12791 12791 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
11-21 20:32:45.922 12787 12787 D AndroidRuntime: Calling main entry com.android.commands.am.Am
11-21 20:32:45.945 12791 12791 I Radio-JNI: register_android_hardware_Radio DONE
11-21 20:32:45.948 12787 12787 D AndroidRuntime: Shutting down VM
11-21 20:32:45.956 12791 12791 D AndroidRuntime: Calling main entry com.android.commands.am.Am
11-21 20:32:45.965 10032 10032 W System.err: android.content.pm.PackageManager$NameNotFoundException
11-21 20:32:45.966 10032 10032 W System.err:    at com.topjohnwu.magisk.container.Policy.<init>(Unknown Source)
11-21 20:32:45.966 10032 10032 W System.err:    at com.topjohnwu.magisk.superuser.SuReceiver.onReceive(Unknown Source)
11-21 20:32:45.966 10032 10032 W System.err:    at android.app.ActivityThread.handleReceiver(ActivityThread.java:3041)
11-21 20:32:45.966 10032 10032 W System.err:    at android.app.ActivityThread.-wrap18(ActivityThread.java)
11-21 20:32:45.966 10032 10032 W System.err:    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1562)
11-21 20:32:45.966 10032 10032 W System.err:    at android.os.Handler.dispatchMessage(Handler.java:102)
11-21 20:32:45.966 10032 10032 W System.err:    at android.os.Looper.loop(Looper.java:154)
11-21 20:32:45.966 10032 10032 W System.err:    at android.app.ActivityThread.main(ActivityThread.java:6121)
11-21 20:32:45.966 10032 10032 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
11-21 20:32:45.966 10032 10032 W System.err:    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
11-21 20:32:45.966 10032 10032 W System.err:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
11-21 20:32:45.968 12791 12791 D AndroidRuntime: Shutting down VM
11-21 20:32:45.983   370 12789 D Magisk  : su: return code: [0]
11-21 20:32:46.002   370 12785 D Magisk  : su: return code: [0]
11-21 20:32:46.020 12797 12797 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 0 <<<<<<
11-21 20:32:46.023 12797 12797 D AndroidRuntime: CheckJNI is OFF
11-21 20:32:46.094 12797 12797 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
11-21 20:32:46.132 12797 12797 I Radio-JNI: register_android_hardware_Radio DONE
11-21 20:32:46.144 12797 12797 D AndroidRuntime: Calling main entry com.android.commands.am.Am
11-21 20:32:46.154 10032 10032 W System.err: android.content.pm.PackageManager$NameNotFoundException
11-21 20:32:46.154 10032 10032 W System.err:    at com.topjohnwu.magisk.container.Policy.<init>(Unknown Source)
11-21 20:32:46.154 10032 10032 W System.err:    at com.topjohnwu.magisk.superuser.SuReceiver.onReceive(Unknown Source)
11-21 20:32:46.154 10032 10032 W System.err:    at android.app.ActivityThread.handleReceiver(ActivityThread.java:3041)
11-21 20:32:46.155 10032 10032 W System.err:    at android.app.ActivityThread.-wrap18(ActivityThread.java)
11-21 20:32:46.155 10032 10032 W System.err:    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1562)
11-21 20:32:46.155 10032 10032 W System.err:    at android.os.Handler.dispatchMessage(Handler.java:102)
11-21 20:32:46.155 10032 10032 W System.err:    at android.os.Looper.loop(Looper.java:154)
11-21 20:32:46.155 10032 10032 W System.err:    at android.app.ActivityThread.main(ActivityThread.java:6121)
11-21 20:32:46.155 10032 10032 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
11-21 20:32:46.155 10032 10032 W System.err:    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
11-21 20:32:46.155 10032 10032 W System.err:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
11-21 20:32:46.157 12797 12797 D AndroidRuntime: Shutting down VM
11-21 20:32:47.386  4736  4736 W WindowManager: Attempted to remove non-existing token: android.os.Binder@bbb54bd
11-21 20:32:47.398  9733 12782 D DuckHunterFragment: su -c '/data/data/com.offsec.nethunter/files/scripts/bootkali duck-hunt-run /opt/duckout.sh'
11-21 20:32:47.459   370 12835 D Magisk  : su: request from client: 23
11-21 20:32:47.459   370 12835 D Magisk  : su: request from uid=[10094] (#2)
11-21 20:32:47.460   370 12835 D Magisk  : su: waiting child: [12836]
11-21 20:32:47.460 12836 12836 D Magisk  : su: child process started
11-21 20:32:47.460 12836 12836 D Magisk  : su: argc=[1]
11-21 20:32:47.461 12836 12836 D Magisk  : su: argv[0]=[su]
11-21 20:32:47.461 12836 12836 D Magisk  : su: cwd=[/]
11-21 20:32:47.461 12836 12836 D Magisk  : su: pts_slave=[]
11-21 20:32:47.461 12836 12836 D Magisk  : su: use namespace of pid=[12834]
11-21 20:32:47.526   370 12838 D Magisk  : su: request from client: 13
11-21 20:32:47.526   370 12838 D Magisk  : su: request from uid=[0] (#3)
11-21 20:32:47.527   370 12838 D Magisk  : su: waiting child: [12839]
11-21 20:32:47.527 12839 12839 D Magisk  : su: child process started
11-21 20:32:47.527 12839 12839 D Magisk  : su: argc=[3]
11-21 20:32:47.527 12839 12839 D Magisk  : su: argv[0]=[su]
11-21 20:32:47.527 12839 12839 D Magisk  : su: argv[1]=[-c]
11-21 20:32:47.528 12839 12839 D Magisk  : su: argv[2]=[/data/data/com.offsec.nethunter/files/scripts/bootkali duck-hunt-run /opt/duckout.sh]
11-21 20:32:47.528 12839 12839 D Magisk  : su: cwd=[/]
11-21 20:32:47.528 12839 12839 D Magisk  : su: pts_slave=[]
11-21 20:32:47.528 12839 12839 D Magisk  : su: use namespace of pid=[12837]
11-21 20:32:47.616   370 12844 D Magisk  : su: request from client: 21
11-21 20:32:47.616   370 12844 D Magisk  : su: request from uid=[0] (#4)
11-21 20:32:47.617   370 12844 D Magisk  : su: waiting child: [12845]
11-21 20:32:47.617 12845 12845 D Magisk  : su: child process started
11-21 20:32:47.617 12845 12845 D Magisk  : su: argc=[4]
11-21 20:32:47.618 12845 12845 D Magisk  : su: argv[0]=[su]
11-21 20:32:47.618 12845 12845 D Magisk  : su: argv[1]=[-c]
11-21 20:32:47.618 12845 12845 D Magisk  : su: argv[2]=[setenforce]
11-21 20:32:47.618 12845 12845 D Magisk  : su: argv[3]=[0]
11-21 20:32:47.618 12845 12845 D Magisk  : su: cwd=[/]
11-21 20:32:47.618 12845 12845 D Magisk  : su: pts_slave=[]
11-21 20:32:47.619 12845 12845 D Magisk  : su: use namespace of pid=[12843]
11-21 20:32:47.643   370 12844 D Magisk  : su: return code: [0]
11-21 20:32:47.877 12840 12840 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 0 <<<<<<
11-21 20:32:47.881 12840 12840 D AndroidRuntime: CheckJNI is OFF
11-21 20:32:47.957 12840 12840 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
11-21 20:32:47.964 12846 12846 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 0 <<<<<<
11-21 20:32:47.967 12846 12846 D AndroidRuntime: CheckJNI is OFF
11-21 20:32:47.992 12840 12840 I Radio-JNI: register_android_hardware_Radio DONE
11-21 20:32:48.004 12840 12840 D AndroidRuntime: Calling main entry com.android.commands.am.Am
11-21 20:32:48.014 10032 10032 W System.err: android.content.pm.PackageManager$NameNotFoundException
11-21 20:32:48.014 10032 10032 W System.err:    at com.topjohnwu.magisk.container.Policy.<init>(Unknown Source)
11-21 20:32:48.014 10032 10032 W System.err:    at com.topjohnwu.magisk.superuser.SuReceiver.onReceive(Unknown Source)
11-21 20:32:48.014 10032 10032 W System.err:    at android.app.ActivityThread.handleReceiver(ActivityThread.java:3041)
11-21 20:32:48.014 10032 10032 W System.err:    at android.app.ActivityThread.-wrap18(ActivityThread.java)
11-21 20:32:48.014 10032 10032 W System.err:    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1562)
11-21 20:32:48.014 10032 10032 W System.err:    at android.os.Handler.dispatchMessage(Handler.java:102)
11-21 20:32:48.014 10032 10032 W System.err:    at android.os.Looper.loop(Looper.java:154)
11-21 20:32:48.014 10032 10032 W System.err:    at android.app.ActivityThread.main(ActivityThread.java:6121)
11-21 20:32:48.014 10032 10032 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
11-21 20:32:48.014 10032 10032 W System.err:    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
11-21 20:32:48.015 10032 10032 W System.err:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
11-21 20:32:48.018 12840 12840 D AndroidRuntime: Shutting down VM
11-21 20:32:48.037 12846 12846 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
11-21 20:32:48.072 12846 12846 I Radio-JNI: register_android_hardware_Radio DONE
11-21 20:32:48.083 12846 12846 D AndroidRuntime: Calling main entry com.android.commands.am.Am
11-21 20:32:48.098 10032 10032 W System.err: android.content.pm.PackageManager$NameNotFoundException
11-21 20:32:48.098 10032 10032 W System.err:    at com.topjohnwu.magisk.container.Policy.<init>(Unknown Source)
11-21 20:32:48.098 10032 10032 W System.err:    at com.topjohnwu.magisk.superuser.SuReceiver.onReceive(Unknown Source)
11-21 20:32:48.098 10032 10032 W System.err:    at android.app.ActivityThread.handleReceiver(ActivityThread.java:3041)
11-21 20:32:48.099 10032 10032 W System.err:    at android.app.ActivityThread.-wrap18(ActivityThread.java)
11-21 20:32:48.099 10032 10032 W System.err:    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1562)
11-21 20:32:48.099 10032 10032 W System.err:    at android.os.Handler.dispatchMessage(Handler.java:102)
11-21 20:32:48.099 10032 10032 W System.err:    at android.os.Looper.loop(Looper.java:154)
11-21 20:32:48.099 10032 10032 W System.err:    at android.app.ActivityThread.main(ActivityThread.java:6121)
11-21 20:32:48.099 10032 10032 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
11-21 20:32:48.099 10032 10032 W System.err:    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
11-21 20:32:48.099 10032 10032 W System.err:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
11-21 20:32:48.101 12846 12846 D AndroidRuntime: Shutting down VM
11-21 20:32:49.087  7799  8341 I PlayCommon: [66] com.google.android.play.a.h.e(248): Preparing logs for uploading
11-21 20:32:49.088  7799  8341 I PlayCommon: [66] com.google.android.play.a.h.e(252): No file ready to send
11-21 20:32:49.403  4736  4736 W WindowManager: Attempted to remove non-existing token: android.os.Binder@2ffeffe
11-21 20:32:51.421  4736  4736 W WindowManager: Attempted to remove non-existing token: android.os.Binder@8e4367b
11-21 20:32:54.734  6917 11526 W io.grpc.internal.dj: [{0}] Failed to resolve name. status={1}
11-21 20:32:55.562   370 12838 D Magisk  : su: return code: [0]
11-21 20:32:55.567   370 12835 D Magisk  : su: return code: [0]

**Comment by j3223** [comment]
Hi, thanks for your work, but after browsing on the internet I apparently found out that there already was an issue like this one and it was fixed with a file on this link. https://github.com/offensive-security/nethunter-app/tree/hidfixes/assets/nh_files/modules I will try to download it and pit it in the correct path. I have searched before opening this issue for similar issues, but didnt find any, but im nevertheless sorry to open this when it already exists as an issue and is already fixed. Thank you for the help, I will cloe this issue a fast as ill manage to pull it of... :)
